File: Flagrum.Core/Gfxbin/Gmdl/MessagePack/GfxbinHeader.cs
using System.Collections.Generic;

namespace Flagrum.Core.Gfxbin.Gmdl.MessagePack;

public class GfxbinHeader
{
    public uint Version { get; set; }
    public Dictionary<string, string> Dependencies { get; set; }
    public IList<ulong> Hashes { get; set; }

    public void Read(MessagePackReader reader)
    {
        Version = reader.Read<uint>();
        Dependencies = reader.Read<Dictionary<string, string>>();
        Hashes = reader.Read<List<ulong>>();
    }
}

File: src/Flagrum.Blender/features/graphics/gfxbinheader.py
from dataclasses import dataclass


@dataclass
class GfxbinHeader:
    version: int
    dependencies: dict[str, str]
    hashes: list[int]

    def __init__(self, reader):
        self.version = reader.read()
        self.dependencies = reader.read()

        hash_count = reader.read()
        self.hashes = []
        for i in range(hash_count):
            self.hashes.append(reader.read())
